model/Hierarchy.php

```diff
@@ -373,7 +393,8 @@ public function markedNodeIDs() {
 
 	/**
 	 * Return an array of this page and its ancestors, ordered item -> root.
-	 * @return array
+	 *
+	 * @return SiteTree[]
```

Comment:
The [PSR](https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc.md#arrays) allows `array` and `SiteTree[]` but only 1 is currently helpful for static analysis (`array`). We shoudl decide which to use:
- `array` with no additions.
- custom collection classes (e.g `SiteTreeList`) which enforce a type on insertion.